src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/MemoryPool.cs

```diff
@@ -60,20 +61,44 @@ public class MemoryPool : IDisposable
         /// </summary>
         private bool _disposedValue = false; // To detect redundant calls
 
+#if DEBUG
+        public MemoryPoolBlock Lease([CallerMemberName] string memberName = "",
+                             [CallerFilePath] string sourceFilePath = "",
+                             [CallerLineNumber] int sourceLineNumber = 0)
+        {
+            Debug.Assert(!_disposedValue, "Block being leased from disposed pool!");
+#else
+
```

Comment:
Can you move this after the doc comments or does the compiler complain about the args?